atom/renderer/lib/override.coffee

```diff
@@ -118,6 +118,12 @@ Object.defineProperty window.history, 'length',
   get: ->
     getHistoryOperation 'length'
 
-# Make document.hidden return the correct value.
+# Make document.hidden and document.visibilityState return the correct value.
 Object.defineProperty document, 'hidden',
-  get: -> !remote.getCurrentWindow().isVisible()
+  get: ->
+    currentWindow = remote.getCurrentWindow()
+    !currentWindow.isFocused() || !currentWindow.isVisible()
```

Comment:
Updated. One doubt though, isnt the situation of non-minimized, non-focused, visible window same as a backgrounded tab ? Shouldnt the value of `document.hidden` be true in this case ?

But yeah, users can always override this with what they see fit as variables, we can just leave it at minimize and visibility, i guess.